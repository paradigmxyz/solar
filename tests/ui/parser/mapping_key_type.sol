type U is int;
enum E {
    A,
    B
}

library L{
    enum E1 {
        A,
        B
    }
}

contract C {
    mapping(uint => uint) m0;
    mapping(E => uint) m1;
    mapping(U => uint) m2;
    mapping(L.E1 => uint) m3;
    mapping(uint[] => uint) m4; //~ ERROR: expected `=>`, found `[`
}
