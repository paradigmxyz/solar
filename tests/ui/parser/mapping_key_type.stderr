error: expected `=>`, found `[`
  --> ROOT/tests/ui/parser/mapping_key_type.sol:LL:CC
   |
LL |     mapping(uint[] => uint) m4;
   |                 ^ expected `=>`

error: aborting due to 1 previous error

